helpers/struct_iterator/struct_iterator.go

```diff
@@ -0,0 +1,60 @@
+package struct_iterator
+
+import (
+	"fmt"
+	"reflect"
+	"unicode"
+)
+
+type StructIterator struct {
+	idx      int
+	numField int
+	val      reflect.Value
+}
+
+func (si *StructIterator) Next() bool {
+	si.idx += 1
+	if si.inRange() && si.isPublic() {
+		return true
+	} else if si.inRange() && !si.isPublic() {
+		return si.Next()
+	}
+	return false
+}
+
+func (si *StructIterator) inRange() bool {
+	return si.idx < si.numField
+}
+
+func (si *StructIterator) ItemName() string {
+	return si.val.Type().Field(si.idx).Name
+}
+
+func (si *StructIterator) ItemValue() interface{} {
+	return si.val.Field(si.idx).Interface()
+}
```

Comment:
Add a new method call is

Tag(string)
https://pkg.go.dev/reflect#StructTag.Get

Reply:
Used Lookup instead for more expected results